### ISSUE TITLE
List View: Remove aria-description in favor of aria-describedby

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55165]
### Description

https://github.com/WordPress/gutenberg/pull/49907 added the ability to render an `aria-description` attribute to the List View.

As pointed out in other issues and PRs, `aria-description` is new in ARIA 1.3, which is not a finalized recommendation yet. As such, we can't use it, yet. We should instead use the traditional method to provide an accessible description:
- `aria-describedby` pointing to a hidden element containing the description.

This is not just an issue about formal compliance. It's about actual support by browsers and assistive technology.
Being a new ARIA 1.3 property, `aria-description` may be supported by some new browser versions and by some new assistive technology versions. However, there is no guarantee of full support. Also, we need to support _old_ combinaitons of browsers and (more importantly) assistive technology.

That said, at the moment one of the known implementation is also buggy, so that's one more reason to not use it.
See https://twitter.com/aardrian/status/1711353551518044626

The cangelog should be updated as well.
Cc: @scruffian @andrewserong @alexstine @joedolson 

### Step-by-step reproduction instructions

See directly in the codebase at:
https://github.com/WordPress/gutenberg/blob/4ce163686d80f51cf97ccfc9b5a1b2873d8c9e11/packages/block-editor/src/components/list-view/index.js#L276

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by joedolson** [comment]
Agreed. Support for  aria-description is a long way from complete, and the specification is incomplete and could change. We need to be using well supported attributes.

**Comment by alexstine** [comment]
This should be an easy refactor if someone wants to try it. If not, I'll take it.

**Comment by scruffian** [comment]
Fix here: [List View: Change the aria-description attribute to aria-describedby](https://github.com/WordPress/gutenberg/pull/55264#top)
